File: java/src/test/java/problems101_200/SymmetricTreeTest.java
package problems101_200;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class SymmetricTreeTest extends SampledTest {
    @Test
    public void test() {
        for (SymmetricTree solution : SymmetricTree.values()) {
            for (SymmetricTreeSample sample : getSamples(SymmetricTreeSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isSymmetric(TreeNode.of(sample.input.root)))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/SymmetricTreeTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class SymmetricTreeTest : SampledTest() {
    @Test
    fun test() =
        SymmetricTree.entries.forEach { solution ->
            getSamples<Array<SymmetricTreeSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.isSymmetric(TreeNode.of(*input.root)))
                    .isEqualTo(output)
            }
        }
}
